sdk/keyvault/keyvault-secrets/review/keyvault-secrets.api.md

```diff
@@ -4,34 +4,54 @@
 
 ```ts
 
+import * as coreHttp from '@azure/core-http';
 import { HttpClient } from '@azure/core-http';
 import { HttpPipelineLogger } from '@azure/core-http';
-import * as msRest from '@azure/core-http';
 import { PagedAsyncIterableIterator } from '@azure/core-paging';
 import { PageSettings } from '@azure/core-paging';
+import { Poller } from '@azure/core-lro';
+import { PollOperationState } from '@azure/core-lro';
 import { RequestOptionsBase } from '@azure/core-http';
 import { ServiceClientOptions } from '@azure/core-http';
 import { TokenCredential } from '@azure/core-http';
 
 // @public
-export interface DeletedSecret extends Secret {
-    readonly deletedDate?: Date;
-    recoveryId?: string;
-    readonly scheduledPurgeDate?: Date;
+export interface DeletedSecret {
+    properties: SecretProperties & {
+        recoveryId?: string;
+        readonly scheduledPurgeDate?: Date;
+        readonly deletedDate?: Date;
+    };
+    value?: string;
 }
 
 // @public
-export type DeletionRecoveryLevel = "Purgeable" | "Recoverable+Purgeable" | "Recoverable" | "Recoverable+ProtectedSubscription";
+export class DeleteSecretPoller extends Poller<DeleteSecretPollOperationState, DeletedSecret> {
+    // Warning: (ae-forgotten-export) The symbol "DeleteSecretPollerOptions" needs to be exported by the entry point index.d.ts
+    constructor(options: DeleteSecretPollerOptions);
+    delay(): Promise<void>;
+    intervalInMs: number;
+}
 
 // @public
-export interface GetSecretOptions {
-    requestOptions?: msRest.RequestOptionsBase;
-    version?: string;
+export interface DeleteSecretPollOperationState extends PollOperationState<DeletedSecret> {
+    // Warning: (ae-forgotten-export) The symbol "SecretClientInterface" needs to be exported by the entry point index.d.ts
+    // 
+    // (undocumented)
+    client: SecretClientInterface;
+    // (undocumented)
+    name: string;
+    // (undocumented)
+    requestOptions?: RequestOptionsBase;
 }
 
 // @public
-export interface GetSecretsOptions {
-    requestOptions?: msRest.RequestOptionsBase;
+export type DeletionRecoveryLevel = "Purgeable" | "Recoverable+Purgeable" | "Recoverable" | "Recoverable+ProtectedSubscription";
```

Comment:
Not sure if it matters, but we had decided against an "enum" for these and to just leave them as strings.

Reply:
this is one of these extensible enums. Typescript won't complain if some other value appears during a live run, or from JavaScript